test/promql/tests/test_network.yaml

```diff
@@ -0,0 +1,123 @@
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+      - series: 'container_network_transmit_bytes_total{namespace="prod", pod="pre-prod-pod"}'
+        values: '0 2 2 2 4 5 5 0 4 4 5'
+      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="", namespace="prod", pod="pre-prod-pod"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+      - series: 'container_network_transmit_bytes_total{namespace="test", pod="test-pod"}'
+        values: '0 2 2 2 4 5 5 7 8 9 10'
+      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="test", namespace="test", pod="test-pod"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+      - series: 'container_network_transmit_bytes_total{namespace="another-test", pod="test-pod"}'
+        values: '0 2 2 2 4 5 5 7 8 9 10'
+      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="test", namespace="another-test", pod="test-pod"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+
+    promql_expr_test:
+
+      # Test the core metric with namespace=prod
+      # Since the sample at 3m is `_` (no sample), the value we get is the last captured value (6)
+      - expr: appstudio_container_network_transmit_bytes_total{namespace="prod"}
+        eval_time: 3m
+        exp_samples:
+        - labels: 'appstudio_container_network_transmit_bytes_total{namespace="prod", pod="prod-pod", label_pipelines_appstudio_openshift_io_type="release"}'
+          value: 6
+        - labels: 'appstudio_container_network_transmit_bytes_total{namespace="prod", pod="pre-prod-pod"}'
+          value: 5
+
+      # Test the core metric with namespace=test
+      - expr: appstudio_container_network_transmit_bytes_total{namespace="test"}
+        eval_time: 4m
+        exp_samples:
+        - labels: 'appstudio_container_network_transmit_bytes_total{namespace="test", pod="test-pod", label_pipelines_appstudio_openshift_io_type="test"}'
+          value: 8
+
+      # Test the core metric with namespace=another-test
+      - expr: appstudio_container_network_transmit_bytes_total{namespace="another-test"}
+        eval_time: 4m
+        exp_samples:
+        - labels: 'appstudio_container_network_transmit_bytes_total{namespace="another-test", pod="test-pod", label_pipelines_appstudio_openshift_io_type="test"}'
+          value: 8
+
+
+      # Test the increase in the last 4 minutes after 5 minutes of samples, filtered by namespace=prod
+      - expr: increase(appstudio_container_network_transmit_bytes_total{namespace="prod"}[4m])
+        eval_time: 5m
+        exp_samples:
+        # We have 2 results:
+        #    one for the results that includes the label_pipelines_appstudio_openshift_io_type="release"
+        #    and one for label_pipelines_appstudio_openshift_io_type=""
+        #
+        # The increase function range is [5m-4m] to [5m]
+        # Since the sample in 5m is `_` we get the last captured value (10)
+        # so the increase between 1 and 10 is 9
```

Comment:
Is the increase calculation made between the base metric and the label metric? (is that where the "1" from?)

Reply:
No, the increase is done on `appstudio_container_network_transmit_bytes_total` which is the base metric `container_network_transmit_bytes_total` with the added `label_pipelines_appstudio_openshift_io_type` label